Lib/dataclasses.py

```diff
@@ -1332,58 +1332,69 @@ class C:
 
 
 def _asdict_inner(obj, dict_factory):
-    if type(obj) in _ATOMIC_TYPES:
+    obj_type = type(obj)
+    if obj_type in _ATOMIC_TYPES:
```

Comment:
This change might make the case for atomic types a _tiny_ bit slower, since we now have to assign to the variable `obj_type` .  But it is faster for all other cases, so seems fine.

Reply:
Yes. This is mitigated by the fact that the first pass through `_asdict_inner` is always the dataclass instance itself, so the `hasattr` speedup (removing a function call) somewhat offsets the obj_type slowdown.